buddyinfo_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procfs
+
+import (
+	"os"
+	"testing"
+)
+
+func TestBuddyInfo(t *testing.T) {
+	file, err := os.Open("fixtures/buddyinfo")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer file.Close()
+
+	buddyInfo, err := parseBuddyInfo(file)
```

Comment:
Let's not use internal package functions in tests, this would have catched your last fix automatically. Instead use the `fs.NewBuddyInfo()` function directly and put the fixtures in different folders, e.g. `FS("fixtures/buddyinfo/short").NewBuddyInfo()`.